data-prepper-plugins/kafka-plugins/src/test/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSourceBufferAccumulatorTest.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.kafka.source;
+
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.kafka.configuration.KafkaSourceConfig;
+import org.opensearch.dataprepper.plugins.kafka.configuration.SchemaConfig;
+import org.opensearch.dataprepper.plugins.kafka.configuration.TopicsConfig;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.doCallRealMethod;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
+@SuppressWarnings("deprecation")
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+class KafkaSourceBufferAccumulatorTest {
+
+	@Mock
+	private KafkaSourceBufferAccumulator<String, String> buffer;
+
+	@Mock
+	private KafkaSourceConfig sourceConfig;
+	
+	@Mock
+	private TopicsConfig topicsConfig;
+	@Mock
+	private KafkaConsumer<Object, Object> kafkaConsumer;
+	
+	@Mock
+	List<TopicsConfig> mockList = new ArrayList<TopicsConfig>();
+	@Mock
+	private SchemaConfig schemaConfig;
+	
+	@Mock
+	private PluginMetrics pluginMetrics;
+	
+	@Mock
+	private Buffer<Record<Object>> record;
+
+	@Mock
+	List<Record<Object>> kafkaRecords;
+
+	@BeforeEach
+	void setUp() throws Exception {
+		when(sourceConfig.getTopics()).thenReturn((mockList));
+		when(mockList.get(0)).thenReturn(topicsConfig);
+		when(sourceConfig.getSchemaConfig()).thenReturn(schemaConfig);
+
+		when(sourceConfig.getSchemaConfig()).thenReturn(mock(SchemaConfig.class));
+
+		buffer = new KafkaSourceBufferAccumulator<>(topicsConfig, sourceConfig, "plaintext", pluginMetrics);
+	}
+
+	@Test
+	void testWriteEventOrStringToBuffer_plaintext_schemaType() throws Exception {
+		createObjectWithSchemaType("plaintext");
+
+		KafkaSourceBufferAccumulator<String, String> spyBuffer = spy(buffer);
+		doCallRealMethod().when(spyBuffer).getEventRecord("anyString");
+		spyBuffer.getEventRecord("anyString");
+		verify(spyBuffer).getEventRecord("anyString");
+		assertNotNull(spyBuffer.getEventRecord("anyString"));
```

Comment:
Assertion should be more specific. It cannot be just a check for not null. You should check the contents of the event record with the expected values.

Reply:
We are working on it and will update this as part of the bug list https://github.com/opensearch-project/data-prepper/issues/2713